### ISSUE TITLE
Regression: superfluous link flags if both --libs and --cflags specified

### ISSUE THREAD

**Issue by ifreund** [pkgconf/pkgconf#275]
Correct output of version 1.8.0:
```
$ pkg-config --libs --cflags wlroots
-I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -lwlroots 
```
Incorrect output of version 1.9.3:
```
$ pkg-config --libs --cflags wlroots
-I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind -lwlroots -lxcb-render -lxcb 
```
Note that if used individually neither `--libs` or `--cflags` outputs `-lxcb-render`/`-lxcb` (also version 1.9.3):
```
$ pkg-config --libs wlroots         
-lwlroots 
$ pkg-config --cflags wlroots
-I/usr/include/libdrm -I/usr/include/pixman-1 -I/usr/include/elogind 
```

Tested against  wlroots 0.15.1 with has the following wlroots.pc file on my void linux system:
```
prefix=/usr
libdir=${prefix}/lib64
includedir=${prefix}/include

have_drm_backend=true
have_x11_backend=true
have_libinput_backend=true
have_xwayland=true
have_gles2_renderer=true
have_vulkan_renderer=true

Name: wlroots
Description: Wayland compositor library
Version: 0.15.1
Requires.private: wayland-server >= 1.20, libdrm >= 2.4.109, gbm >= 17.1.0, xkbcommon, libudev, pixman-1, egl, glesv2, vulkan >= 1.2.182, libinput >= 1.14.0, xcb, xcb-dri3, xcb-present, xcb-render, xcb-renderutil, xcb-shm, xcb-xfixes, xcb-xinput, wayland-client, libseat >= 0.2.0, xcb-composite, xcb-icccm, xcb-res, xcb-errors
Libs: -L${libdir} -lwlroots
Libs.private: -lm -lrt
Cflags: -I${includedir}
```

**Comment by ifreund** [comment]
From skimming the issue tracker this seems potentially related to the changes in #258, haven't found time to look at the code yet though.

**Comment by kaniini** [comment]
Fixed in e2a12fd4c2d1c93cc7b1d76ee07c51a39fb73cd2.  We now reset the solver when solving for `--libs`, since we require different behavior.

**Comment by ifreund** [comment]
Thanks for the fix!
